### ISSUE TITLE
Massive slowdown when multiple lua_parse or lua_bar used.

### ISSUE THREAD

**Issue by y-salnikov** [brndnmtthws/conky#967]
**Issue**
After recent update of Arch Linux, the conky become very slow, about one refresh per 10 seconds. While testing different configs, i find out that slowdowns occur when lua functions are used. 
**Information**
I think it may be not a conkys bug, because conky was not updated during system update and the older versions behave same way.
```strace``` output spammed with messages like ```futex(0x55bc568b03c0, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, FUTEX_BITSET_MATCH_ANY) = 0```

[my test config](https://github.com/brndnmtthws/conky/files/4592646/conky.zip)

**Comment by y-salnikov** [comment]
Find out that version 1.11.3 works just fine and version 1.11.4 doesn't.

**Comment by y-salnikov** [comment]
Bisect shows that culprit is commit c352069b31b2dbb4d87503603d6c6d3ddd517df6 "Add support for nested |exec| expressions"

**Comment by tflori** [comment]
I also have a massive problem with the upgrade. Since then each call to `conky_parse` adds about 0.1 seconds to my lua function.

For the new version I have to put everything in one `conky_parse` call - and I'm not sure that this is the solution I want to go..

**Comment by tflori** [comment]
@brndnmtthws what do we do here now? do I have to downgrade manually to 1.11.3 or do you accept a revert of this new feature?  it would be nice if any maintainer could say something to this issue.

**Comment by philer** [comment]
> For the new version I have to put everything in one conky_parse call

I also noticed `conky_parse` being a major bottle neck (1.11.5) and did just that, with an "eager loading" helper for convenience. If you're interested you can check the code out [here](https://github.com/philer/polycore/blob/master/src/data.lua#L38).

**Comment by tflori** [comment]
interesting idea. but nevertheless it is just a workaround: 1 call to lua_parse / conky_parse takes more time than it took before for all my calls (roughly about 40 calls).

**Comment by gottfriedm** [comment]
This slowdown of `conky_parse()` is really annoying, please can this be fixed? It makes the lua scripts utterly useless.

This code
```
tt = socket.gettime()
print(conky_parse("${hwmon 1 temp 1}"), conky_parse("${hwmon 1 temp 3}"), conky_parse("${hwmon 1 temp 2}"))
print('time: ',socket.gettime()-tt)
```

with conky-1.10.8
```
35      37      30
time:   0.00016593933105469
35      37      30
time:   0.00017213821411133
35      37      30
time:   0.00016498565673828
35      37      30
time:   0.00016403198242188
35      37      30
time:   0.00017499923706055
35      37      30
time:   0.00015807151794434
```

Same with conky-1.11.6
```
35      37      30
time:   0.30080986022949
35      37      30
time:   0.30092906951904
35      37      30
time:   0.30434203147888
36      37      30
time:   0.30092811584473
35      37      30
time:   0.30187606811523
35      37      30
time:   0.30196404457092
35      37      30
time:   0.30118894577026
```

It is more than 1000 times slower!

**Comment by powerman** [comment]
@tflori Can you please describe your workaround in more details? E.g. how to optimize something like this:
```
${color}CPU1${alignr}${lua_parse prefix_lt ${hwmon 1 temp 2} 75 ${color0} ${color3}}°
${color}CPU2${alignr}${lua_parse prefix_lt ${hwmon 1 temp 3} 75 ${color0} ${color3}}°
${color}CPU3${alignr}${lua_parse prefix_lt ${hwmon 1 temp 4} 75 ${color0} ${color3}}°
${color}CPU4${alignr}${lua_parse prefix_lt ${hwmon 1 temp 5} 75 ${color0} ${color3}}°
```

**Comment by tflori** [comment]
@powerman I don't have a workaround. The workaround is from @philer  It basically greps all data with one call to conky_parse.

**Comment by tflori** [comment]
@brndnmtthws can we please get an update to this issue? I'm still sticking to v1.11.3 because of this issue.

**Comment by plikhari** [comment]
I created a workaround in 2015. You can see I have a multi-conky setup - you can say that each conky is running a widget of sorts. Yes, that is a fully functional mathematically correct Astrolabe - similar to the Prague one.

- We tweak the update interval `conky_info.update_interval` - store this value in a tmp file.
- Update the current update interval - `conky_set_update_interval(0.1)` to 0.1 second.

Once the script is up and running in 2 to 3 seconds, you reset the update interval to its original value you stored in that tmp file.

You will never notice any lag ...

![conkywx_2021-07-13_10-39-19](https://user-images.githubusercontent.com/14849928/125397001-f64be500-e3ca-11eb-8786-ce37631aa6c4.png)

**Comment by tflori** [comment]
I don't see a connection between multiple conky instances and the slowdown of lua_parse and conky_parse in 1.11.4 (a version from 2020). A process that is slow from beginning might not be affected this much by this issue but you see the test result from @gottfriedm: three calls to conky_parse to get the temps from 3 sensors is 100 times slower with the new version. You can't update this info every 0.1 seconds as it takes about 0.3 seconds to get the info.

**Comment by gottfriedm** [comment]
It seems that one of the newer conky versions fixes the issue, see updates [here](https://github.com/brndnmtthws/conky/issues/1031#issuecomment-1087192339). If others can confirm that the problem is gone, then we can close these issues.

**Comment by y-salnikov** [comment]
Fixed in 1.12.2.
